op-devstack/sysgo/l2_el_synctester.go

```diff
@@ -0,0 +1,128 @@
+package sysgo
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/ethereum-optimism/optimism/op-devstack/devtest"
+	"github.com/ethereum-optimism/optimism/op-devstack/shim"
+	"github.com/ethereum-optimism/optimism/op-devstack/stack"
+	"github.com/ethereum-optimism/optimism/op-devstack/stack/match"
+	"github.com/ethereum-optimism/optimism/op-service/client"
+	sttypes "github.com/ethereum-optimism/optimism/op-sync-tester/synctester/backend/types"
+)
+
+// SyncTesterEL is an L2ELNode implementation that runs a sync tester service.
+// It provides RPC endpoints that can be used by CL nodes for testing sync functionality.
+type SyncTesterEL struct {
+	mu sync.Mutex
+
+	id      stack.L2ELNodeID
+	l2Net   *L2Network
+	jwtPath string
+	authRPC string
+	userRPC string
+
+	// Sync tester specific fields
+	clNodeID stack.L2CLNodeID
+	fcuState sttypes.FCUState
+	p        devtest.P
+
+	// Reference to the orchestrator to find the EL node to connect to
+	orch *Orchestrator
+}
+
+var _ L2ELNode = (*SyncTesterEL)(nil)
+
+func (n *SyncTesterEL) hydrate(system stack.ExtensibleSystem) {
+	require := system.T().Require()
+	rpcCl, err := client.NewRPC(system.T().Ctx(), system.Logger(), n.userRPC, client.WithLazyDial())
+	require.NoError(err)
+	system.T().Cleanup(rpcCl.Close)
+
+	l2Net := system.L2Network(stack.L2NetworkID(n.id.ChainID()))
+	sysL2EL := shim.NewL2ELNode(shim.L2ELNodeConfig{
+		RollupCfg: l2Net.RollupConfig(),
+		ELNodeConfig: shim.ELNodeConfig{
+			CommonConfig: shim.NewCommonConfig(system.T()),
+			Client:       rpcCl,
+			ChainID:      n.id.ChainID(),
+		},
+		ID: n.id,
+	})
+	sysL2EL.SetLabel(match.LabelVendor, "sync-tester")
+	l2Net.(stack.ExtensibleL2Network).AddL2ELNode(sysL2EL)
+}
+
+func (n *SyncTesterEL) Start() {
+	n.mu.Lock()
+	defer n.mu.Unlock()
+
+	// The SyncTesterEL should connect to the existing sync tester service
+	// Get the endpoint from the orchestrator's syncTester service
+	if n.orch.syncTester == nil || n.orch.syncTester.service == nil {
+		n.p.Logger().Error("syncTester service not available in orchestrator")
+		return
+	}
+
+	// Use NewEndpoint to get the correct session-specific endpoint for this chain ID
+	syncTesterEndpoint := n.orch.syncTester.service.NewEndpoint(n.id.ChainID())
+
+	n.userRPC = syncTesterEndpoint
+	// For sync tester, the engine RPC should include FCU state as query parameters
+	n.authRPC = fmt.Sprintf("%s?latest=%d&safe=%d&finalized=%d",
+		syncTesterEndpoint, n.fcuState.Latest, n.fcuState.Safe, n.fcuState.Finalized)
+}
+
+func (n *SyncTesterEL) Stop() {
+	// The SyncTesterEL is just a proxy, so there's nothing to stop
```

Comment:
Idea: We can stop the service by implementing sync tester's sync API, and explicitly make the request to the endpoint not served.